slack-message.c

```diff
@@ -514,11 +514,6 @@ static void get_history_cb(SlackAccount *sa, gpointer data, json_value *json, co
 }
 
 void slack_get_history(SlackAccount *sa, SlackObject *conv, const char *since, unsigned count) {
-	if (SLACK_IS_CHANNEL(conv)) {
-		SlackChannel *chan = (SlackChannel*)conv;
```

Comment:
Is there a reason to remove this? I think it's necessary for the blist get history callback to work on on a closed chat.

Reply:
Ah, I was wondering what the reason was! But yes, this is definitely necessary, because if this call is not removed, every chat opens on every login, regardless of whether there are any new messages.

But I still don't fully understand why the call is necessary to have here in the first place, because this code path eventually leads into `handle_message`, which calls the exact same thing.

I can do some testing in this area, but I might not have time for it immediately.